---
title: Methods for testing publication bias in ecological and evolutionary meta-analyses
author: Shinichi Nakagawa, Malgorzata Lagisz, Michael D. Jennions, Julia Koricheva, Daniel W.A. Noble, Timothy H. Parker, Alfredo Sanchez-Tojar, Yefeng Yang, Rose O'Dea
date: '`r format(Sys.time(), "%d %B, %Y")`'
output:
  html_document:
    code_download: yes
    code_folding: hide
    depth: 5
    toc: yes
    toc_float: yes
    toc_depth: 5
  pdf_document:
    toc: yes
subtitle: Supplementary Information - Appendix S1-S5 including figure code
bibliography: references.bib
biblio-style: "apalike"
csl: jane.csl
link-citations: yes
---

```{r setup, include = FALSE}
# knitr setting
knitr::opts_chunk$set(
  #message = FALSE,
  warning = FALSE, # no warnings
  tidy = TRUE,
  cache = TRUE
)

# cleaning up
rm(list = ls())

## numbers >= 10^5 will be denoted in scientific notation, and rounded to 2 digits
options(digits=3)
# clearing up
```

```{r, eval =FALSE}
# install the orchaRd package
install.packages("devtools")
install.packages("tidyverse")
install.packages("metafor")
install.packages("patchwork")
install.packages("R.rsp")

devtools::install_github("itchyshin/orchard_plot", subdir = "orchaRd", force = TRUE, build_vignettes = TRUE)

# install the dmetar pacakge
if (!require("devtools")) {
  install.packages("devtools")
}
devtools::install_github("MathiasHarrer/dmetar")

# getting metaAidR
install.packages("devtools")
library(devtools)
install_github("daniel1noble/metaAidR")

```


```{r function, echo = FALSE, eval = TRUE}
# required packages
pacman::p_load(tidyverse,
               metafor,
               gt,
               pander,
               stringr,
               openxlsx,
               rotl,
               ape,
               kableExtra,
               patchwork,
               here,
               lme4,
               readxl,
               orchaRd,
               patchwork,
               cowplot,
               metaviz,
               meta,
               ggplotify,
               grid,
               dmetar,
               viridis,
               metaAidR
)

# necessary custom functions

# extracting mean and CI from each metafor model
estimates.CI <- function(model){
  db.mf <- data.frame(model$b,row.names = 1:nrow(model$b))
  db.mf <- cbind(db.mf,model$ci.lb,model$ci.ub,row.names(model$b))
  names(db.mf) <- c("mean","lower","upper","estimate")
  return(db.mf[,c("estimate","mean","lower","upper")])
}

# r and Zr transfromation can be done by tanh and atanh - base function

```

## Appendix S1: Survey of publication bias tests reported in ecology and evolution meta-analyses

*Ask Rose to finish this off* - references are not done etc - need fixing - use  (ask Yefeng and Alfredo do help as well) -- e.g. [@viechtbauer2010conducting]

### S1.1: Survey method

The main objective of the literature survey was to capture the types of publication bias methods recently used in the field of ecology and evolution. The survey reported here was conducted alongside a larger survey on reporting standards from the PRISMA-EcoEvo project (Preferred Reporting Items for Systematic reviews and Meta-Analyses in Ecology and Evolutionary Biology; full details and materials provided in {O’Dea, 2019 #28}). The survey was based on 102 meta-analysis papers published between 1 January 2010 and 25 March 2019. We aimed for the sample of 102 meta-analyses to be representative of recently published meta-analyses in ecology and evolutionary biology journals.

#### S1.1.1 Representative Sample

To select the representative sample of meta-analysis publications, we first searched the Scopus database for papers with (“meta-analy\*” OR “metaanaly\*” OR “meta-regression”) in the title, abstract, or keywords (where the asterisk allows for any ending to the word, such as “meta-analyses” or “meta-analysis”), restricted the date to papers published after 2009, and restricted the ISSN (International Standard Serial Number) to journals classified as ‘Ecology’ and/or ‘Evolutionary Biology’ by the 2017 rankings of the ISI InCites Journal Citation Reports. On 25 March 2019 this search returned 1,668 papers from 134 journals.

Second, the returned papers were categorised as being from journals listed as ‘Ecology’, ‘Evolution’, or ‘Both’, and we reduced the number of journals and papers. To reduce the number of journals we arranged journals in descending order of frequency (i.e. to indicate which journals published the most meta-analyses). After removing journals in more applied sub-fields (such as ecology economics), we retained the top 10 journals classified as ecology, the top 10 journals classified as evolutionary biology, and the top 11 journals classified as both ecology and evolutionary biology. The list of 31 retained journals is shown in Table S1. Next, we selected 297 studies to be screened for inclusion in the sample, by using the ‘sample’ function in R (v. 3.5.1; R Core Team, 2018) to randomly select up to 17 studies from the journals classified as ‘Evolutionary Biology’ (57 studies total) or ‘Both’ (150 studies total), and up to 9 studies from journals classified as ‘Ecology’ (90 studies total). 

##### Table S1.1
Journals screened in our search for a representative sample of meta-analyses published in ecology and evolutionary biology. ISI Classification is based on the 2017 ISI InCites Journal Citation Reports. The number of papers returned from the Scopus database search is described by ‘N hits’, while the number of studies selected for screening is described by ‘N screened’.
```{r}
# getting the data and formatting some variables (turning character vectors to factors)
read_excel(here("data/TableS1.xlsx"), na = "NA") %>% 
  #mutate_if(is.character, as.factor) %>%  
  kbl() %>%
  kable_paper() %>% 
  kable_styling("striped", position = "left")
```

#### S1.1.2 Inclusion criteria and screening methods
The sample of 102 meta-analysis papers met the following four inclusion criteria: (1) the study addressed a question in the fields of ecology and evolutionary biology; (2) claimed to present results from a meta-analysis; (3) performed a search for, and collected, data from the primary literature; (4) used a statistical model to analyse effect sizes that were collected from multiple studies. Paper screening was conducted in two stages. First, two authors (RO and ML) conducted parallel abstract screening. Conflicting decisions were discussed and resolved. Second, 64% of screened abstracts underwent parallel full-text screening by RO and ML, in consultation with SN.


#### S1.1.3 Assessing 102 meta-analysis papers
Papers were independently assessed by seven authors (RO, DN, JK, MJ, ML, RO, SN, and TP) as part of a larger, time-consuming survey. The one survey question pertaining to this project was: “Which publication bias tests are reported in the paper? (Select all that apply)”. There were 11 possible choices for respondents to select: (A) Funnel plots (including contour-enhanced funnel plots); (B) Normal quantile (QQ) plots (Wang & Bushman); (C) Correlation-based tests (e.g. Begg & Manzumdar rank correlation); (D) Regression-based tests (e.g. Egger regression and its variants); (E) File drawer numbers or fail-safe N (Rosenthal, Orwin or Rosenberg method); (F) Trim-and-fill tests; (G) P-curve, P-uniform or its variants; (H) Selection (method) models (e.g. Copas, Hedges or lyengar & Greenhouse model); (I) Time-lag bias tests (e.g., regression or correlation on the relationship between effect sizes and time or cumulative meta-analysis); (J) None reported and (K) ‘Other’ methods. According to Sutton ({, 2009 #17}) (see also {Vevea, 2019 #16}), Methods A-D are tests detecting publication bias, whereas Methods E-F are assessing the impact of publication bias. 

### S1.2 Results
Among the 102 assessed papers, 17.8% did not report any tests of publication bias. 
Most meta-analysis papers reported one or more tests of publication bias (17.8% of assessed papers did not include any assessment of publication bias). These results suggest tests of publication bias have become more common in recent year in ecology and evolutionary biology, as over half of older meta-analyses assessed by Nakagawa and Santos ({, 2012 #19}) and Koricheva and Gurevitch ({Koricheva, 2014 #18}) did not report any tests of publication bias (although our results are not directly comparable, due to different survey methods). Still, inferential tests of publication bias remain uncommon. By far the most popular test of publication bias were funnel plots (32.4%; Table S2), with all remaining methods represented by fewer than 15% of papers. All methods except ‘selection models’ were present in at least one paper (with ‘other’ being selected for a weighted histogram used by {Loydi, 2013 #70}; Table S2). The absence of selection model methods could be because these methods are comparatively technically challenging, or because ecologists and evolutionary biologists are not yet aware of the benefits of these methods.

##### Table S1.2
Frequency with which publication bias tests were reported in the 102 meta-analysis publications, ranked in order of decreasing popularity. No tests were reported for 17.80% of papers.

```{r}
# getting the data and formatting some variables (turning character vectors to factors)
read_excel(here("data/TableS2.xlsx"), na = "NA") %>% 
  #mutate_if(is.character, as.factor) %>%  
  kbl() %>%
  kable_paper() %>% 
  kable_styling("striped", position = "left")
```

## Appendix S2: Equivalence between Equations 8 and 9

As in the main text, Equation 8 is as follows
$$
z_{i} = \beta_{0} + \beta_{1}prec_i + e_i,\\
$$
where $e_{i} \sim \mathcal{N}(0, \sigma_e^2)$. We can rewrite this as:

$$
y_{i}/se_i = \beta_{0} + \beta_{1}(1/se_i) + e_i,\\
$$
If we multiple both side by $se_i$, we have:

$$
y_{i} = \beta_{0}se_i + \beta_{1} + e_{i}se_{i},
$$

which is basically the same, if $\beta_0$ and $\beta_1$ are swapped, as:

$$
y_{i} = \beta_{0} + \beta_{1}se_i + \epsilon_i,
$$
where $\epsilon_i \sim \mathcal{N}(0, v_i\phi)$ as in Equation 9. We can show this using a simulated data as well (we use the data used for Figure 3 and 4). Below the first result is from Equation 8 and the second Equation 9:


```{r}
# creating data - the same data as Figures 3 + 4
set.seed(77777)
# setting parameters
n.effect <- 100
sigma2.s <- 0.05
beta1 <- 0.2

# using negative binomial we get good spread of sample size
n.sample <- rnbinom(n.effect, mu = 30, size = 0.7) + 4

# variance for Zr
vi <- 1/(n.sample - 3)

# moderator x 1
xi <- rnorm(n.effect)

# there is underling overall effect to r = 0.2 or Zr = 0.203
Zr <- atanh(0.2) + beta1*xi + rnorm(n.effect, 0, sqrt(sigma2.s)) + rnorm(n.effect, 0, sqrt(vi))
#qplot(Zr, 1/sqrt(vi))

# data frame
dat <- data.frame(yi = Zr, vi = vi, sei = sqrt(vi), xi = xi, ni = n.sample, prec = 1 / sqrt(vi), wi = 1/vi, zi = Zr/sqrt(vi))
rows <- 1:nrow(dat)
expected <- which(1/dat$sei < 5 & dat$yi < 0.25)
unexpected <- which(1/dat$sei > 4.7 & dat$yi > 0.25)
col_point <- ifelse(rows %in% expected, "red", ifelse(rows %in% unexpected, "blue", "black"))
dat$col_point <- col_point

# data with "publication bias"
dat2 <- dat[dat$col_point != "red", ]

# they are the same
eq_8 <- lm(zi ~ prec, data = dat2)
eq_9 <- lm(yi ~ sei, weight = wi, data = dat2)
```

```{r}
summary(eq_8)
summary(eq_9)
```

As you can see above, Equation 8's intercept is identical the slope of Equation 9 whiile Equation 8's slope to Equation 9's intercept. Note that corresponding standard errors (SE) are also the same. 


## Appendix S3: Faile-safe *N* comparisions

We use the function `fsn()` in the R package, `metafor`, as in this web page  ([link](https://wviechtb.github.io/metafor/reference/fsn.html)).

```{r}
# an example data using risk ratio (not response ratio)
dat <- escalc(measure="RR", ai=tpos, bi=tneg, ci=cpos, di=cneg, data=dat.bcg)
fsn(yi, vi, data=dat, type="Rosenthal")
# setting effect size targe as -0.1
fsn(yi, data=dat, type="Orwin", target = 0.1)
fsn(yi, vi, data=dat, type="Rosenberg")
```

## Appendix S4: Mulitlevel meta-regession method for publication bias

In this section, we provide worked examples showing how to test for publication bias in meta-analytic datasets with high heterogeneity and several laters of non-independence, including phylogenetic non-independence, which are common features of meta-analytic datasets in ecology and evolution (Senior et al. 2016, Noble et al. 2017). We used two openly available meta-analytic databases to show how to implement our suggested multilevel meta-regression method (see main text) on meta-analytic databases with correlation effect sizes (**Zr**: section S4.1) and effect sizes based on mean differences between two groups (SMD, lnRR: section S4.2).

### S4.1: Correlation effect sizes (**Zr**)

```{r, results="hide"}
# importing dataset with correlation coefficients
dataset.r.original <- read.xlsx(here("data","ft044.xlsx"), colNames = TRUE, sheet = 1)

# transforming r to Zr for the analyses, and estimating VZr for the weighting
dataset.r.original$Sample.size <- as.numeric(dataset.r.original$Sample.size)

# subset dataset to those with 4 or more individuals (this removes two rows, one without sample size and one with sample size = 3)
dataset.r.original <- dataset.r.original[dataset.r.original$Sample.size>3 & !(is.na(dataset.r.original$Sample.size)),]

dataset.r.original$yi <- atanh(dataset.r.original$Correlation)
dataset.r.original$vi <- 1/(as.numeric(dataset.r.original$Sample.size)-3) #the effect based on 3 individuals won't be able to be analyzed because VZr is then infinite (see VZr function above to understand). Also, there is a missing sample size. 

# renaming some variables to make it easier
dataset.r.original$study <- dataset.r.original$Reference
dataset.r.original <- dataset.r.original[ , !names(dataset.r.original) %in% c("Reference")]
```

For our first worked example, we use the dataset provided by Garamszegi et al. (2012), who tested the hypothesis that behaviours are correlated (behavioural syndrome) across studies and species, which they found support for. Here, we reanalyse the data from Garamszegi et al. (2012) by first conducting a phylogenetic multilevel intercept-only meta-analytic model and then testing for evidence of publication bias following the approaches outlined in the main text.

#### S4.1.1 - Phylogenetic relationships

Since multiple species (n = `r length(unique(dataset.r.original$Species))` species) are included in this dataset, we need to account for phylogenetic non-independence in our statistical models (Chamberlain et al. 2012, Cinar et al. 2020). For that, we first search for the the species names in the Open Tree Taxonomy (Rees and Cranston 2017) to confirm and correct species names, then build a phylogenetic tree for all the species included in the database (Figure S1) by retrieving their phylogenetic relationships from the Open Tree of Life (Hinchliff et al. 2015) using the R package `rotl` (Michonneau et al. 2016). We then estimate branch lengths following Grafen (1989) as implemented in the function 'compute.brlen()' of the R package `ape` (Paradis and Schliep 2019). Finally, we construct a phylogenetic relatedness correlation matrix that will be fitted as part of the random effect structure of our models (see below).

```{r}
# fixing a typo in the original list of species names
dataset.r.original$Species <- ifelse(dataset.r.original$Species=="Acrochephalus schoenobaenus",
                                     "Acrocephalus schoenobaenus",
                                     dataset.r.original$Species)

# updating a name that was not being well recognized by the Open Tree Taxonomy
dataset.r.original$Species <- ifelse(dataset.r.original$Species=="Eurotestudo boettgeri",
                                     "Testudo boettgeri",
                                     dataset.r.original$Species)

# fixing another typo in the original list of species names
dataset.r.original$Species <- ifelse(dataset.r.original$Species=="Taenopygia guttata",
                                     "Taeniopygia guttata",
                                     dataset.r.original$Species)

# # obtaining dataframe listing the Open Tree identifiers potentially matching our list of species (be aware that this will take a few minutes, and you can load the data below)
#taxa <- tnrs_match_names(names = unique(dataset.r.original$Species))

# # saving the taxonomic data created on the 18th of February 2021 to speed the process in the future and make the code fully reproducible if taxonomic changes are implemented in the future
# save(taxa,file = "taxa_Open_Tree_of_Life_20210218.RData")

# loading the taxonomic data (taxa) created on the 18th of February 2021
load(here("data", "taxa_Open_Tree_of_Life_20210218.RData"))

# # check approximate matches
# taxa[taxa$approximate_match==TRUE,]
# 
# # check synonyms matches
# taxa[taxa$is_synonym==TRUE,]
# 
# # check number of matches
# taxa[taxa$number_matches>1,]

# # some further checks
# ott_id_tocheck <- taxa[taxa$number_matches != 1,"ott_id"]

# for(i in 1:length(ott_id_tocheck)){
#   print(inspect(taxa, ott_id = ott_id_tocheck[i]))
# }

# all phylogenetic data seems in order now

# however, the ott_id for Parma unifasciata cannot be found when retrieving the phylogenetic relationships, so to trick this we are going to use the ott_id of another Parma species, Parma oligolepis (ott_id = 323186), which in this case it is fine because we only include two species belonging to the genus Parma, and therefore, the phylogenetic relationship will be the same for our purposes
# tnrs_match_names(names = 'Parma oligolepis')
# taxa[taxa$unique_name=="Parma unifasciata","ott_id"] <- 323186

# # retrieving phylogenetic relationships among taxa in the form of a trimmed sub-tree (you can simply load the tree at the bottom)
# tree <- tol_induced_subtree(ott_ids = taxa[["ott_id"]], label_format = "name")

# # we need to check for the existence of polytomies
# is.binary.tree(tree) # No polytomies, so we can proceed.

# to confirm that our tree covers all the species we wanted it to include, and make sure that the species names in our database match those in the tree, we use the following code

# tree$tip.label <- gsub("_"," ", tree$tip.label)
# intersect(as.character(tree$tip.label), as.character(dataset.r.original$Species))
# setdiff(as.character(dataset.r.original$Species), as.character(tree$tip.label)) #listed in our database but not in the tree
# setdiff(as.character(tree$tip.label),as.character(dataset.r.original$Species)) # listed in the tree but not in our database

# All but Pan and Parma oligolepis are the same species, the "problem" is that synonyms have been used in the tree. We are going to leave all the names as in Open Tree of Life except Pan and Parma oligolepis, which we are going to substitute by Pan troglodytes and Parma unifasciata

# # we start by fixing the following names in the tree
# tree$tip.label[tree$tip.label=="Pan"]<-"Pan troglodytes"
# tree$tip.label[tree$tip.label=="Parma oligolepis"]<-"Parma unifasciata"

# setdiff(as.character(dataset.r.original$Species), as.character(tree$tip.label)) #listed in our database but not in the tree
# setdiff(as.character(tree$tip.label),as.character(dataset.r.original$Species)) # listed in the tree but not in our database


# changing the names in our database to follow those in the tree. We are creating a new Species.updated variable so that it is clear that this list of Species is an updated version compared to the original one
dataset.r.original$Species.updated <- dataset.r.original$Species

dataset.r.original$Species.updated <- ifelse(dataset.r.original$Species.updated=="Carduelis chloris",
                                             "Chloris chloris",
                                             dataset.r.original$Species.updated)

dataset.r.original$Species.updated <- ifelse(dataset.r.original$Species.updated=="Dendroica pensylvaniaca",
                                             "Setophaga pensylvanica",
                                             dataset.r.original$Species.updated)

dataset.r.original$Species.updated <- ifelse(dataset.r.original$Species.updated=="Sylvia melanocephala",
                                             "Curruca melanocephala",
                                             dataset.r.original$Species.updated)

# we are also updating the original Species variable since we have to fit both Species and Species.updated in our models (see below)
dataset.r.original$Species <- dataset.r.original$Species.updated

# setdiff(as.character(dataset.r.original$Species.updated), as.character(tree$tip.label)) #listed in our database but not in the tree
# setdiff(as.character(tree$tip.label),as.character(dataset.r.original$Species.updated)) # listed in the tree but not in our database

# all in order

# we can now save the tree
#save(tree, file = "tree_20211802.Rdata")

dataset.r <- dataset.r.original

# load the saved tree (tree)
load(here("data","tree_20211802.Rdata")) 
```


```{r}
# compute branch lengths of tree
phylo_branch <- compute.brlen(tree, method = "Grafen", power = 1)

# # check tree is ultrametric
# is.ultrametric(phylo_branch) # TRUE

# matrix to be included in the models
phylo_cor <- vcv(phylo_branch, cor = T)
```

##### Figure S4.1

Phylogenetic tree of all species included in this meta-analytic dataset. Notice that some of the names shown in the tree correspond to the most updated synonyms according to the Open Tree Taxonomy (Rees and Cranston 2017) of those originally avialable in the dataset of Garamszegi et al. (2012).

<br/><br/>
```{r, fig.height = 9, fig.width = 9.5, fig.align = "center"}
# we can then plot the tree
plot(tree, type = "fan", cex=0.65, label.offset =.05, no.margin = TRUE) #check: https://www.rdocumentation.org/packages/ape/versions/5.3/topics/plot.phylo
```

<br/><br/>

#### S4.1.2: Exploratiory data analysis

Before running any meta-analytic model, it is important to explore the meta-analytic dataset in search of potential outliers that could represent data extraction errors. To do so, we can draw funnel plots showing the standard error and the inverse of the standard error (i.e. precision) in the y-axes (Figure S4.2). From these plots, we can conclude that no outliers seem to be present in this meta-analytic dataset.

##### Figure S4.2

Funnel plots with SE (left-hand side) and precision (1/SE; right-hand side) as measures of uncertainty to explore the existence of outliers in the meta-analytic dataset.

<br/><br/>

```{r}
par(mfrow = c(1, 2))
funnel(dataset.r$yi, dataset.r$vi, yaxis="sei",
       #xlim = c(-3, 3),
       xlab = "Effect size (Zr)") 
funnel(dataset.r$yi, dataset.r$vi, yaxis="seinv",
       #xlim = c(-3, 3),
       xlab = "Effect size (Zr)") 
```

<br/><br/>

#### S4.1.3: Mulilevel-meta-analysis

Now that we have created a matrix with the phylogenetic relationships among species and have confirmed that no outliers seem to be present, we can then use the database from Garamszegi et al. (2012) to provide a worked example on how to test of publication bias using the multilevel meta-regression method suggested in the main text. First, let's have a look at the detailed results of a multilevel phylogenetic meta-analytic (intercept-only) model (Table S4.1), including the corresponding funnel plot showing the overall effect or meta-analytic mean (Figure S4.3).

```{r}
# creating a unit-level random effect to model residual variance in metafor
dataset.r$obsID <- 1:nrow(dataset.r)

# running multilevel intercept-only meta-analytic model
meta.analysis.model.r <- rma.mv(yi, vi,
                                mods=~1,
                                random=list(~ 1 | Species.updated, # phylo effect
                                            ~ 1 | Species, # non-phylo effect
                                            ~ 1 | study, 
                                            ~ 1 | obsID), 
                                R = list(Species.updated = phylo_cor), #phylogenetic relatedness
                                method="REML",
                                test="t", # using t dist rather than z because t is a better, more conservative distribution than z
                                data=dataset.r)

# print(meta.analysis.model.r, digits=3)

# extracting the mean, 95% confidence intervals and 95% prediction intervals
metaanalytic.mean.model.r <- predict(meta.analysis.model.r, digits=3)

# estimating I2 as measure of heterogeneity
I2.model.r <- i2_ml(meta.analysis.model.r, method = "wv")

# creating a data.frame with the meta-analytic mean and heterogeneity estimates
table.model.r <- data.frame(n=length(unique(dataset.r$study)),
                            k=nrow(dataset.r),
                            mean=round(metaanalytic.mean.model.r[[1]],2),
                            CI=paste0("[",round(metaanalytic.mean.model.r[[3]],2),",",round(metaanalytic.mean.model.r[[4]],2),"]"),
                            PI=paste0("[",round(metaanalytic.mean.model.r[[5]],2),",",round(metaanalytic.mean.model.r[[6]],2),"]"),
                            I2_obsID=round(I2.model.r[["I2_obsID"]]*100,1),
                            I2_paperID=round(I2.model.r[["I2_study"]]*100,1),
                            I2_nonphylo=round(I2.model.r[["I2_Species"]]*100,1), 
                            I2_phylo=round(I2.model.r[["I2_Species.updated"]]*100,1),
                            I2_total=round(I2.model.r[["I2_total"]]*100,1)
)

# creating a fancy table using the R package 'gt'
table.model.r.gt <- table.model.r %>% 
  gt() %>% 
  cols_label(n=md("**n**"),
             k=md("**k**"),
             mean=md("**Meta-analytic mean**"),
             CI=md("**95% CI**"),
             PI=md("**95% PI**"),
             I2_obsID=md("***I*<sup>2</sup><sub>residual</sub>\n(%)**"),
             I2_paperID=md("***I*<sup>2</sup><sub>study</sub>\n(%)**"),
             I2_nonphylo=md("***I*<sup>2</sup><sub>non-phylogeny</sub>\n(%)**"),
             I2_phylo=md("***I*<sup>2</sup><sub>phylogeny</sub>\n(%)**"),
             I2_total=md("***I*<sup>2</sup><sub>total</sub>    \n(%)**"),
  ) %>%
  cols_align(align = "center") %>%
  tab_source_note(source_note = md("n = number of studies; k = number of effects; CI = confidence interval; PI = prediction interval; *I*<sup>2</sup> = heterogeneity")) #%>%
# tab_options(table.width=770)

table.model.r.gt

```

**Table S4.1.** Results of the phylogenetic multilevel intercept-only meta-analysis testing the relationship between behaviours across species. Estimates are presented as standardized effect sizes using Fisher’s transformation (i.e. **Zr**).


##### Figure S4.3

Funnel plot using precision (1/SE) as a measure of uncertainty and showing the meta-analytic mean (vertical line) obtained in the phylogenetic multilevel intercept-only meta-analysis.

<br/><br/>

```{r}
funnel(meta.analysis.model.r, yaxis="seinv",
       xlab = "Effect size (Zr)")
```

<br/><br/>

#### S4.1.4: Publication bias tests with multilevel meta-regression

##### S4.1.4.1: Meta-regression with SE (uni-moderator)

To test for publication bias, we can first fit a simple phylogenetic multilevel meta-regression to explore whether there is some evidence of small-study effects in our meta-analytic data. To do so, we fit a uni-moderator phylogenetic multilevel meta-regression including the effect sizes' standard errors (SE or sei) as the only moderator (see Equation 21 from the main text). This meta-regression will provide some information about the existence of small-study effects (i.e. asymmetry in the distribution of effect sizes), but importantly, the best evidence for small-study effects will come from the all-in publication bias test (multi-moderator meta-regression) described in section S4.1.4.3, since this multi-moderator meta-regression will test whether asymmetry exists effects after accouting for the heterogeneity explained by all included moderators (more in main text).

```{r}
# creating a variable for the standard error of each effect size (i.e. the square root of the sampling variance, see Figure 1 from the main text)
dataset.r$sei <- sqrt(dataset.r$vi)

# Application of Equation 21 from the main text
publication.bias.model.r.se <- rma.mv(yi, vi,
                                      mod =~1+sei,
                                      random=list(~ 1 | Species.updated, # phylo effect
                                                  ~ 1 | Species, # non-phylo effect
                                                  ~ 1 | study, 
                                                  ~ 1 | obsID), 
                                      R = list(Species.updated = phylo_cor), #phylogenetic relatedness
                                      method="REML",
                                      test="t", # using t dist rather than z because t is a better, more conservative distribution than z
                                      data=dataset.r)

# print(publication.bias.model.r.se,digits=3)

# extracting the mean and 95% confidence intervals
estimates.publication.bias.model.r.se <- estimates.CI(publication.bias.model.r.se)
```

According to this uni-moderator meta-regression, there is some indication of small-study effects since the slope of the moderator 'SE' is marginally statistically significant (**slope = `r round(estimates.publication.bias.model.r.se[2,2],2)`, 95% CI = [`r round(estimates.publication.bias.model.r.se[2,3],2)`,`r round(estimates.publication.bias.model.r.se[2,4],2)`]**; *R<sup>2</sup><sub>marginal</sub>* = `r round(r2_ml(publication.bias.model.r.se)[[1]]*100,1)`%) and shows that effect sizes with larger SE tend to be larger. Nonetheless, as mentioned above, we will confirm the existence of small-study effects after accounting for some of the heterogeneity present in the data using the all-in publication bias test (multi-moderator meta-regression) described in section S4.1.4.3.

##### S4.1.4.2: Meta-regression with `year` (uni-moderator)

To test for time-lag bias, also called decline effects, we can first fit a simple phylogenetic multilevel meta-regression. To do so, we fit a uni-moderator phylogenetic multilevel meta-regression including the year of publication (mean-centred) as the only moderator (see Equation 23 from the main text). The estimated slope for 'year' will provide some evidence on whether effect sizes have become smaller over time since the first effect size was published, however, as above, the best evidence for such declines effects will come from the all-in publication bias test (multi-moderator meta-regression) described in section S4.1.4.3, since this multi-moderator meta-regression will test for the existence of decline effects after accouting for the heterogeneity explained by all included moderators (more in main text).

```{r}
# creating a variable with the year of publication
dataset.r$year <- as.numeric(str_extract(dataset.r$study, "(\\d)+"))

# mean-centring year of publication to help with interpretation, particularly in S4.1.4.3.
dataset.r$year.c <- as.vector(scale(dataset.r$year, scale = F))

# Application of Equation 23 from the main manuscript
publication.bias.model.r.timelag <- rma.mv(yi, vi,
                                           mods=~1+year.c,
                                           random=list(~ 1 | Species.updated, # phylo effect
                                                       ~ 1 | Species, # non-phylo effect
                                                       ~ 1 | study, 
                                                       ~ 1 | obsID), 
                                           R = list(Species.updated = phylo_cor), #phylogenetic relatedness
                                           method="REML",
                                           test="t", # using t dist rather than z because t is a better, more conservative distribution than z
                                           data=dataset.r)

# summary(publication.bias.model.r.timelag)

# extracting the mean and 95% confidence intervals
estimates.publication.bias.model.r.timelag <- estimates.CI(publication.bias.model.r.timelag)
```

According to this uni-moderator meta-regression, there is no indication of decline effects since the slope of the moderator 'year' is essentially zero (**slope = `r round(estimates.publication.bias.model.r.timelag[2,2],2)`, 95% CI = [`r round(estimates.publication.bias.model.r.timelag[2,3],2)`,`r round(estimates.publication.bias.model.r.timelag[2,4],2)`]**; *R<sup>2</sup><sub>marginal</sub>* = `r round(r2_ml(publication.bias.model.r.timelag)[[1]]*100,1)`%). Nonetheless, as mentioned above, we need to confirm this pattern after accounting for some of the heterogeneity present in the data using the all-in publication bias test (multi-moderator meta-regression) described in section S4.1.4.3.

##### S4.1.4.3: All-in publication bias test (multi-moderator)

When heterogeneity exists, which will normally be the case, particularly so in ecology and evolution (Senior et al. 2016), it is best to combine Equations 21 and 23 with other moderators since those additional moderators will generally explain or be predicted to explain some of the heterogeneity among effect sizes. That is, this all-in publication bias test (multi-moderator meta-regression) would be the best test of small-study (publication bias) and decline effects (time-lag bias) in most meta-analytic datasets. For our data, we will run a multi-moderator phylogenetic multilevel meta-regression including the effect sizes' standard errors, the year of publication (mean-centred) and a moderator originally included in Garamszegi et al. (2012), 'captivity status'. Although Garamszegi et al. (2012) included other moderators in their model (i.e. species, taxonomic class, spatial overlap, age, sex and season), we decided to focus only on those moderators that were available for the entire database (see Table 2 in Garamszegi et al. 2012), and we also excluded 'species' and 'taxonomic class' since those effects would be already modelled by our phylogenetic approach.

```{r}
# Application of Equation 24 from the main manuscript

# Note that we are removing the intercept in this model (using '-1') just because the intercept is not of interest at this point and instead we prefer to see the slopes of the continues moderators and the estimates of all the levels of the categorical moderator 'CaptivityC'. Note that whether to remove or not the intercept does not change the conclusions of the small-study effects test or the decline effects test.
publication.bias.model.r.all.se <- rma.mv(yi, vi,
                                          mods=~-1+ # -1 removes the intercept
                                            sei+
                                            year.c+
                                            CaptivityC,   
                                          random=list(~ 1 | Species.updated, # phylo effect
                                                      ~ 1 | Species, # non-phylo effect
                                                      ~ 1 | study, 
                                                      ~ 1 | obsID), 
                                          R = list(Species.updated = phylo_cor), #phylogenetic relatedness
                                          method="REML",
                                          test="t", # using t dist rather than z because t is a better, more conservative distribution than z
                                          data=dataset.r)

#summary(publication.bias.model.r.all.se)

# extracting the mean and 95% confidence intervals
estimates.publication.bias.model.r.all.se <- estimates.CI(publication.bias.model.r.all.se)
```

In this case, the all-in publication bias test confirms what we observed in the uni-moderator meta-regressions above (sections S4.1.4.1 and S4.1.4.2). First, the multi-moderator meta-regression shows a statistically significant slope for the moderator 'SE' (**slope = `r round(estimates.publication.bias.model.r.all.se[1,2],2)`, 95% CI = [`r round(estimates.publication.bias.model.r.all.se[1,3],2)`,`r round(estimates.publication.bias.model.r.all.se[1,4],2)`]**; Figure S4.4), showing evidence of small-study effect. In other words, the largest effect sizes in the dataset tend to be those with the lowest precision (i.e. larger uncertainty; Figure S4.4). This result agrees with the findings originally reported in Garamszegi et al. (2012), where the authors found evidence of publication bias after applying a simple correlation method proposed by Begg and Mazumdar (1994) and the trim-and-fill method (Duval and Tweedie 2000) - although bear in mind that we do not recommend to use those simple methods for meta-analytic databases with multiple levels of non-independence and high heterogeneity (more in the main text). Second, the all-in publication bias test also confirms that there is no evidence of decline effects in the data since the slope of the moderator 'year' was again indistinguishable from zero (**slope = `r round(estimates.publication.bias.model.r.all.se[2,2],2)`, 95% CI = [`r round(estimates.publication.bias.model.r.all.se[2,3],2)`,`r round(estimates.publication.bias.model.r.all.se[2,4],2)`]**; Figure S4.5). Overall, the moderators included in this multi-moderator meta-regression explained a total of `r round(r2_ml(publication.bias.model.r.all.se)[[1]]*100,1)`% (i.e. *R<sup>2</sup><sub>marginal</sub>*) of the heterogeneity observed in the meta-analytic model (Table S4.1).

```{r}
# to plot the 'sei' effect we are simply including the moderator 'CaptivityC' as a random effect for simplicity of plotting. Note that this does not substantially change the estimates of the moderators, and that alternative approaches could be used for this.
publication.bias.model.r.all.se.plot <- rma.mv(yi, vi,
                                               mods=~1+
                                                 sei+
                                                 year.c,
                                               random=list(~1 | CaptivityC,
                                                           ~ 1 | Species.updated, # phylo effect
                                                           ~ 1 | Species, # non-phylo effect
                                                           ~ 1 | study, 
                                                           ~ 1 | obsID), 
                                               R = list(Species.updated = phylo_cor), #phylogenetic relatedness
                                               method="REML",
                                               test="t", # using t dist rather than z because t is a better, more conservative distribution than z
                                               data=dataset.r)

# I was unsure whether this residual plot was necessary and did not really know how to best accommodate it into the text. Do you want to add it? 
# meta.regression.r.residuals.funnel.plot <- rma.mv(yi, vi,
#                                                   mods=~1+
#                                                     year.c+
#                                                     CaptivityC,
#                                                   random=list(~ 1 | Species.updated, # phylo effect
#                                                               ~ 1 | Species, # non-phylo effect
#                                                               ~ 1 | study, 
#                                                               ~ 1 | obsID), 
#                                                   R = list(Species.updated = phylo_cor), #phylogenetic relatedness
#                                                   method="REML",
#                                                   test="t", # using t dist rather than z because t is a better, more conservative distribution than z
#                                                   data=dataset.r)
# 
# funnel(meta.regression.r.residuals.funnel.plot, yaxis = "seinv")


```

##### Figure S4.4

Effect sizes with larger SE tend to be larger, providing evidence of small-study effects in this meta-analytic dataset. The solid line represents the model estimate and shading shows the 95% confidence intervals.

```{r fig.height = 5.5, fig.width = 7.5, fig.align = "center"}
# predicting effect sizes for a range of SE from the minimum observed in the data to the maximum observed in data, while year is kept at its mean value, in this case 0 since year was mean-centred
predict.publication.bias.model.r.all.se.plot.1 <- predict(publication.bias.model.r.all.se.plot,
                                                          newmods=cbind(seq(min(dataset.r$sei),
                                                                            max(dataset.r$sei),
                                                                            0.005),
                                                                        c(0)))

newdat <- data.frame(sei=seq(min(dataset.r$sei),
                             max(dataset.r$sei),
                             0.005),
                     fit=predict.publication.bias.model.r.all.se.plot.1$pred,
                     upper=predict.publication.bias.model.r.all.se.plot.1$ci.ub,
                     lower=predict.publication.bias.model.r.all.se.plot.1$ci.lb,
                     stringsAsFactors=FALSE)

xaxis <- dataset.r$sei
yaxis <- dataset.r$yi

plot(xaxis,yaxis,
     type="n",
     ylab="",
     xlab="",
     xaxt="n",
     yaxt="n",
     ylim=c(-2.25,2.25),
     xlim=c(0,1))

abline(a=0,b=0, lwd=1, lty=1)

axis(1,at=seq(0,1,0.1),
     cex.axis=0.8,tck=-0.02)

axis(2,
     at=round(seq(-2.5,2.25,0.5),1),
     cex.axis=0.8,las=2,tck=-0.02)

title(xlab = "standard error (SE)", 
      ylab = "effect size (Zr)",
      line = 2.75, cex.lab=1.4)

points(xaxis,yaxis,
       bg=rgb(0,0,0, 0.1),
       col=rgb(0,0,0, 0.2),
       pch=21,
       cex=1)

lines(newdat$sei, newdat$fit, lwd=2.75,col="darkorchid4") 

polygon(c(newdat$sei,rev(newdat$sei)),
        c(newdat$lower,rev(newdat$upper)),
        border=NA,col=rgb(104/255,34/255,139/255, 0.5))
```

<br/><br/>

##### Figure S4.5

The overall effect size has remained seemingly unchanged over time since the first effect size was published. The solid line represents the model estimate, shading shows the 95% confidence intervals and individual effect sizes are scaled by their precision (1/SE).

```{r fig.height = 5.5, fig.width = 7.5, fig.align = "center"}
# predicting effect sizes for a range of years from the minimum observed in the data to the maximum observed in data, while SE is kept at its mean value. Keep in mind that alternative ways could be used, for example, one could use median-centring rather than mean-centring.
predict.publication.bias.model.r.all.se.plot.2 <- predict(publication.bias.model.r.all.se.plot,
                                                          newmods=cbind(mean(dataset.r$sei),
                                                                        seq(min(dataset.r$year.c),
                                                                            max(dataset.r$year.c),
                                                                            0.25)))

newdat <- data.frame(year=seq(min(dataset.r$year),
                              max(dataset.r$year),
                              0.25),
                     fit=predict.publication.bias.model.r.all.se.plot.2$pred,
                     upper=predict.publication.bias.model.r.all.se.plot.2$ci.ub,
                     lower=predict.publication.bias.model.r.all.se.plot.2$ci.lb,
                     stringsAsFactors=FALSE)

xaxis <- dataset.r$year
yaxis <- dataset.r$yi
cex.study <- (1/dataset.r$sei)/4

plot(xaxis,yaxis,
     type="n",
     ylab="",
     xlab="",
     xaxt="n",
     yaxt="n",
     ylim=c(-2.25,2.25),
     xlim=c(min(dataset.r$year),max(dataset.r$year)))

abline(a=0,b=0, lwd=1, lty=1)

axis(1,at=seq(min(dataset.r$year),max(dataset.r$year),5),
     cex.axis=0.8,tck=-0.02)

axis(2,
     at=round(seq(-2.5,2.25,0.5),1),
     cex.axis=0.8,las=2,tck=-0.02)

title(xlab = "year of publication", 
      ylab = "effect size (Zr)",
      line = 2.75, cex.lab=1.4)

points(xaxis,yaxis,
       bg=rgb(0,0,0, 0.1),
       col=rgb(0,0,0, 0.2),
       pch=21,
       cex=cex.study)

lines(newdat$year, newdat$fit, lwd=2.75,col="darkorchid4") 

polygon(c(newdat$year,rev(newdat$year)),
        c(newdat$lower,rev(newdat$upper)),
        border=NA,col=rgb(104/255,34/255,139/255, 0.5))
```

<br/><br/>

Once we have confirmed that there is evidence of small-study effects (see above), the next step in our suggested two-step approach is to calculate the adjusted overall effect size, that is, the overall effect once we account for small-study effects (see the main text). Bear in mind that, if the slope for the moderator 'SE' in the all-in publication bias test shown above was not statistically significant, then the best estimate for the adjusted overall effect size would be the *intercept* of that all-in publication bias test (Stanley and Doucouliagos 2012, 2014; more in the main text). However, since the slope for the moderator 'SE' was statistically significant, the approach to estimate the adjusted overall effect is to fit a similar multi-moderator regression to the one above but including the effect sizes' sampling variances (SV or vi) instead of their SE as a moderator. The reason for using sampling variance instead of standard errors when calculating the adjusted overall effect when the slope for 'SE' is statistically signficant is becuase using 'SV' leads to an adjusted overall effect that is less downwardly biased than when using 'SE' (more in the main text; Stanley and Doucouliagos 2012, 2014).

What moderators to include in this version of the all-in publication bias test depends on what we want the adjusted overall effect size to reflect. We are going to fit three different models to showcase some of these possibilities. Importantly, we should keep in mind that publication bias tests should be treated as sensitivity analyses to test the robustness of results from the original analysis (Noble et al., 2017) and that none of the estimated adjusted overall effects should be treated as the real overall value if publication bias did not exist; after all, we can never be sure of what has not been published. The three models that we are going to fit are:

1. Multi-moderator meta-regression only including sampling variance and year of publication (mean-centred) as moderators. The intercept of this meta-regression would represent the adjusted overall effect size. 

2. Multi-moderator meta-regression including sampling variance and year of publication (mean-centred) as moderators, and 'captivity status', which has five levels, as a random effect. The goal of this meta-regression is to provide an adjusted overall effect size after accouting for the variance explained by 'captivity status'. Bear in mind that other alternative approaches might be used (e.g. mean-centring the categorical factor following Schielzeth 2010), but results should not change substantially.

3. Multi-moderator meta-regression including sampling variance, year of publication (mean-centred) and the categorical factor 'captivity status', which has five levels, as moderators. In this meta-regression, we are going to remove the intercept so that we obtain adjusted overall effects for each of the levels of the factor 'captivity status'.

```{r}
# model 1: ignoring CaptivityC
publication.bias.model.r.v.1 <- rma.mv(yi, vi,
                                       mods=~1+vi + 
                                         year.c,
                                       random=list(~ 1 | Species.updated, # phylo effect
                                                   ~ 1 | Species, # non-phylo effect
                                                   ~ 1 | study, 
                                                   ~ 1 | obsID), 
                                       R = list(Species.updated = phylo_cor), #phylogenetic relatedness
                                       method="REML",
                                       test="t", # using t dist rather than z because t is a better, more conservative distribution than z
                                       data=dataset.r)

# model 2: adding CaptivityC as a random effect to provide an adjusted overall effect across levels
publication.bias.model.r.v.2 <- rma.mv(yi, vi,
                                       mods=~1+vi +
                                         year.c, # take Class - that is already a part of phylo
                                       random=list(~ 1 | Species.updated, # phylo effect
                                                   ~ 1 | Species, # non-phylo effect
                                                   ~ 1 | study,
                                                   ~ 1 | obsID,
                                                   ~ 1 | CaptivityC),
                                       R = list(Species.updated = phylo_cor), #phylogenetic relatedness
                                       method="REML",
                                       test="t", # using t dist rather than z because t is a better, more conservative distribution than z
                                       data=dataset.r)

# model 3: providing an adjusted effect for each level of CaptivityC
publication.bias.model.r.v.3 <- rma.mv(yi, vi,
                                       mods=~-1+vi + 
                                         year.c +
                                         CaptivityC, # take Class - that is already a part of phylo   
                                       random=list(~ 1 | Species.updated, # phylo effect
                                                   ~ 1 | Species, # non-phylo effect
                                                   ~ 1 | study, 
                                                   ~ 1 | obsID), 
                                       R = list(Species.updated = phylo_cor), #phylogenetic relatedness
                                       method="REML",
                                       test="t", # using t dist rather than z because t is a better, more conservative distribution than z
                                       data=dataset.r)

# model 3b: comparing results without correcting for publication bias
publication.bias.model.r.v.3b <- rma.mv(yi, vi,
                                        mods=~-1+
                                          CaptivityC, # take Class - that is already a part of phylo   
                                        random=list(~ 1 | Species.updated, # phylo effect
                                                    ~ 1 | Species, # non-phylo effect
                                                    ~ 1 | study, 
                                                    ~ 1 | obsID), 
                                        R = list(Species.updated = phylo_cor), #phylogenetic relatedness
                                        method="REML",
                                        test="t", # using t dist rather than z because t is a better, more conservative distribution than z
                                        data=dataset.r)

# extracting the mean and 95% confidence intervals
estimates.publication.bias.model.r.v.1 <- estimates.CI(publication.bias.model.r.v.1)
estimates.publication.bias.model.r.v.2 <- estimates.CI(publication.bias.model.r.v.2)
estimates.publication.bias.model.r.v.3 <- estimates.CI(publication.bias.model.r.v.3)
estimates.publication.bias.model.r.v.3b <- estimates.CI(publication.bias.model.r.v.3b)
```

Model 1 estimated an adjusted overall effect size that was slightly smaller (**adjusted effect = `r round(estimates.publication.bias.model.r.v.1[1,2],2)`, 95% CI = [`r round(estimates.publication.bias.model.r.v.1[1,3],2)`,`r round(estimates.publication.bias.model.r.v.1[1,4],2)`]**) than that obtained by the multilevel meta-analysis (**unadjusted effect = `r round(metaanalytic.mean.model.r[[1]],2)`, 95% CI = [`r round(metaanalytic.mean.model.r[[3]],2)`,`r round(metaanalytic.mean.model.r[[4]],2)`]**; more in section S4.1.3), highlithing the downward correction of the overall effect after accounting for publication bias. For our worked example, model 2 provides the same exact adjusted overall effect (**adjusted effect><sub>captivityl</sub>  = `r round(estimates.publication.bias.model.r.v.2[1,2],2)`, 95% CI = [`r round(estimates.publication.bias.model.r.v.2[1,3],2)`,`r round(estimates.publication.bias.model.r.v.2[1,4],2)`]**) than model 1 since the variance explained by the random effect 'captivity status' is zero. The latter results agree with the low variance explained by 'captivity status' when included as a moderator (*R<sup>2</sup><sub>marginal</sub>* = `r round(r2_ml(publication.bias.model.r.v.3b)[[1]]*100,1)`%). Ideally, the moderators included in the all-in publication bias test would explain more variance, but this will depend on the specific meta-analytic database being analyzed. Last, model 3 also showed slightly smaller effects for all levels of the categorical factor 'captivity status' (Table S4.2). 

```{r}
# creating a table to show the results of model 2
table.comparing.captivity.levels <- merge(estimates.publication.bias.model.r.v.3[c(3:7),],
                                          estimates.publication.bias.model.r.v.3b,
                                          by="estimate",
                                          all.x=T)

# rounding estimates
table.comparing.captivity.levels <- table.comparing.captivity.levels %>% mutate_at(vars(mean.x, lower.x,upper.x,mean.y,lower.y,upper.y), funs(round(., 2)))


table.comparing.captivity.levels <- data.frame(captivity.level = table.comparing.captivity.levels[,1],
                                               adjusted.mean=table.comparing.captivity.levels[,2],               adjusted.CI=paste0("[",table.comparing.captivity.levels[,3],",",table.comparing.captivity.levels[,4],"]"),
                                               unadjusted.mean=table.comparing.captivity.levels[,5],                                               unadjusted.CI=paste0("[",table.comparing.captivity.levels[,6],",",table.comparing.captivity.levels[,7],"]"))

table.comparing.captivity.levels[,1] <- c("FC","FW","WCT","WCT & FC", "WCT & FW")

# creating a fancy table using the R package 'gt'
table.model.r.captivity.gt <- table.comparing.captivity.levels %>% 
  gt() %>% 
  cols_label(captivity.level=md("**Captivity status**"),
             adjusted.mean=md("**Adjusted mean**"),
             adjusted.CI=md("**Adjusted 95% CI**"),
             unadjusted.mean=md("**Unadjusted mean**"),
             unadjusted.CI=md("**Unadjusted 95% CI**")) %>%
  cols_align(align = "center")

table.model.r.captivity.gt
```

**Table S4.2.** Adjusted and unadjusted for publication bias results for all levels of the moderator 'captivity status' using the all-in publication bias test (multi-moderator meta-regression). Estimates are presented as standardized effect sizes using Fisher’s transformation (i.e. **Zr**).

### S4.2: Mean differences between two groups (lnRR and SMD)

#### S4.2.2: Exploratiory data analysis

as in the section above

```{r, fig.width=7, fig.height= 8}
# importing dataset with mean differences
dataset.original.mean.diff<-read_excel(here("data", "ft027.xlsx"), col_names = TRUE)

#names(dataset.original.mean.diff)
#str(dataset.original.mean.diff)


# lnRR
dataset.rr  <- escalc(measure = "ROM",
                      m1i = T_mean,
                      m2i = C_mean,
                      sd1i = T_sd,
                      sd2i = C_sd,
                      n1i = T_N,
                      n2i = C_N,
                      data = dataset.original.mean.diff,
                      append = T)

dataset.smd <- escalc(measure = "SMD",
                      m1i = T_mean,
                      m2i = C_mean,
                      sd1i = T_sd,
                      sd2i = C_sd,
                      n1i = T_N,
                      n2i = C_N,
                      data = dataset.original.mean.diff,
                      append = T)


# metrics <- data.frame(RR = lnRR$yi, VRR = lnRR$vi, SMD = SMD$yi, VSMD = SMD$vi)
# dataset.mean.diff <- cbind(dataset.original.mean.diff, metrics) # bind_cols()

# clean NA
dataset.rr <- dataset.rr[!is.na(dataset.rr$yi) & dataset.rr$vi != 0, ]

dataset.smd <- dataset.smd[!is.na(dataset.smd$yi) & dataset.smd$vi != 0, ]


# here it is OK - but we find some strange data in SMD and lnRR 
par(mfrow = c(3, 2))
funnel(dataset.rr$yi, dataset.rr$vi, yaxis="seinv",
       #xlim = c(-3, 3),
       ylab = "Precision (1/SE)",
       xlab = "Effect size (lnRR)") 
funnel(dataset.smd$yi, dataset.smd$vi, yaxis="seinv",
       #xlim = c(-3, 3),
       ylab = "Precision (1/SE)",
       xlab = "Effect size (SMD)") 

dataset.rr <- dataset.rr[dataset.rr$vi != min(dataset.rr$vi), ]

dataset.smd <- dataset.smd[dataset.smd$yi != min(dataset.smd$yi),]

# taking outilers
funnel(dataset.rr$yi, dataset.rr$vi, yaxis="seinv",
       #xlim = c(-3, 3),
       ylab = "Precision (1/SE)",
       xlab = "Effect size (lnRR)") 
funnel(dataset.smd$yi, dataset.smd$vi, yaxis="seinv",
       #xlim = c(-3, 3),
       ylab = "Precision (1/SE)",
       xlab = "Effect size (SMD)") 

# calculating "effective sample size" to account for unbalanced sampling, for SMD and lnRR (esz, hereafter, see Equation 25 from the main manuscript)

dataset.rr$e_n <- 4*(dataset.rr$C_N*dataset.rr$T_N) / (dataset.rr$C_N + dataset.rr$T_N)
dataset.smd$e_n <- 4*(dataset.smd$C_N*dataset.smd$T_N) / (dataset.smd$C_N + dataset.smd$T_N)

#using effecitve sampling size

funnel(dataset.rr$yi, dataset.rr$vi, ni = dataset.rr$e_n, yaxis="ni",
       #xlim = c(-3, 3),
       ylab = "Effective sample size",
       xlab = "Effect size (lnRR)") 
funnel(dataset.smd$yi, dataset.smd$vi, ni = dataset.smd$e_n, yaxis="ni",
       #xlim = c(-3, 3),
       ylab = "Effective sample size",
       xlab = "Effect size (SMD)") 

```

Write something how different funnel plots look quite different!

#### S4.2.2: Meta-analytic model

<!--XXX et al. (XXXX) tested the hypothesis that X and Y were correlated across studies and species. Indeed, the authors found support for their hypothesis, showing that X and Y were correlated across studies and species. Here, we reanalysed the data from XXX et al. (XXXX) by conducting a multilevel intercept-only meta-analytic model. Then, we used the data from XXX et al. (XXXX) to provide a worked example on how to test of publication biases in datasets with several layers of non-independence and high heterogeneity (more in the main text), which is common for meta-analyses in ecology and evolution (e.g. Senior et al. 2016; Noble et al. 2017). Detailed results of the meta-analysis are shown in Table S1.-->

```{r}
# creating a unit-level random effect to model residual variance in metafor
dataset.rr$obsID <- 1:nrow(dataset.rr)
dataset.smd$obsID <- 1:nrow(dataset.smd)

# running multilevel intercept-only meta-analytic model
meta.analysis.model.RR <- rma.mv(yi, vi,
                                 mods = ~ 1,
                                 random = list(~ 1 | paperID, 
                                               ~ 1 | obsID),
                                 method = "REML",
                                 test = "t", # using t dist rather than z because t is a better, more conservative distribution than z
                                 data = dataset.rr)


meta.analysis.model.SMD <- rma.mv(yi, vi,
                                  mods = ~ 1,
                                  random = list(~ 1 | paperID, 
                                                ~ 1 | obsID),
                                  method = "REML",
                                  test = "t", # using t dist rather than z because t is a better, more conservative distribution than z
                                  data = dataset.smd)

# extracting the mean, 95% confidence intervals and 95% prediction intervals
#print(meta.analysis.model.1, digits=3)
metaanalytic.mean.model.RR <- predict(meta.analysis.model.RR, digits=3)
metaanalytic.mean.model.SMD <- predict(meta.analysis.model.SMD, digits=3)
#forest(meta.analysis.model.1)

# estimating relative heterogeneity I2
I2.model.RR <- i2_ml(meta.analysis.model.RR)*100
I2.model.SMD <- i2_ml(meta.analysis.model.SMD)*100

# and absolute heterogeneity Q
#Q.model.RR <- c(meta.analysis.model.RR$QE)
#Q.model.SMD <- c(meta.analysis.model.SMD$QE)

# creating a table to show the heterogeneity estimates
table.model.RR <- data.frame(n=length(unique(dataset.rr$paperID)),
                             k=nrow(dataset.rr),
                             mean=round(metaanalytic.mean.model.RR[[1]],2),
                             CI=paste0("[",round(metaanalytic.mean.model.RR[[3]],2),",",round(metaanalytic.mean.model.RR[[4]],2),"]"),
                             PI=paste0("[",round(metaanalytic.mean.model.RR[[5]],2),",",round(metaanalytic.mean.model.RR[[6]],2),"]"),
                             I2_obsID=round(I2.model.RR[[3]],1),
                             I2_paperID=round(I2.model.RR[[2]],1),
                             I2_total=round(I2.model.RR[[2]],1))

rownames(table.model.RR) <- NULL

#write.csv(table.model.RR, "./table/table.model.RR.csv", row.names = FALSE)

table.model.SMD <- data.frame(n=length(unique(dataset.smd$paperID)),
                              k=nrow(dataset.smd),
                              mean=round(metaanalytic.mean.model.SMD[[1]],2),
                              CI=paste0("[",round(metaanalytic.mean.model.SMD[[3]],2),",",round(metaanalytic.mean.model.SMD[[4]],2),"]"),
                              PI=paste0("[",round(metaanalytic.mean.model.SMD[[5]],2),",",round(metaanalytic.mean.model.SMD[[6]],2),"]"),
                              I2_obsID=round(I2.model.SMD[[3]],1),
                              I2_paperID=round(I2.model.SMD[[2]],1),
                              I2_total=round(I2.model.SMD[[1]],1))

rownames(table.model.SMD) <- NULL

#write.csv(table.model.SMD, "./table/table.model.SMD.csv", row.names = FALSE)

# creating a fancy table using the R package 'gt'
table.model.RR.gt <- table.model.RR %>% 
  gt() %>% 
  cols_label(n=md("**n**"),
             k=md("**k**"),
             mean=md("**Meta-analytic mean**"),
             CI=md("**95% CI**"),
             PI=md("**95% PI**"),
             I2_obsID=md("***I*<sup>2</sup><sub>residual</sub> (%)**"),
             I2_paperID=md("***I*<sup>2</sup><sub>study</sub> (%)**"),
             I2_total=md("***I*<sup>2</sup><sub>total</sub> (%)**")) %>%
  cols_align(align = "center") %>%
  tab_source_note(source_note = md("n = number of studies; k = number of effects; CI = confidence interval; PI = prediction interval; *I*<sup>2</sup> = heterogeneity")) #%>%
#tab_options(table.width=775)

table.model.RR.gt


table.model.SMD.gt <- table.model.SMD %>% 
  gt() %>% 
  cols_label(n=md("**n**"),
             k=md("**k**"),
             mean=md("**Meta-analytic mean**"),
             CI=md("**95% CI**"),
             PI=md("**95% PI**"),
             I2_obsID=md("***I*<sup>2</sup><sub>residual</sub> (%)**"),
             I2_paperID=md("***I*<sup>2</sup><sub>study</sub> (%)**"),
             I2_total=md("***I*<sup>2</sup><sub>total</sub> (%)**")) %>%
  cols_align(align = "center") %>%
  tab_source_note(source_note = md("n = number of studies; k = number of effects; CI = confidence interval; PI = prediction interval; *I*<sup>2</sup> = heterogeneity"))  #%>%
#tab_options(table.width=775)

table.model.SMD.gt

```

**Table S1.** Results of the multilevel intercept-only meta-analysis testing the relationship between XXX and YYY across species. Estimates are presented as standardized effect sizes using log transformed ratio of means (i.e., lnRR).standardized mean difference (i.e., Hedges' g)

<br/><br/>

**We could also provide an orchard plot of the results. The reason why I did not do it is because I need to update R to do that and have not found the time. If you'd like to provide an orchard plot, maybe the fastest would be for Yefeng to do that**

```{r}

```


#### S4.2.3: Publication bias tests with mulitlevel meta-regression

Something here?

##### S4.2.3.1: Meta-regression with effective sampling size (uni-moderator)

Here try not repeat what you already said - you can pretty much say as above and what's differet

<!--To test for small-study effects, we are going to follow the two-step approach that we suggested in the main text. First, we run a multilevel meta-regression that has the same random effect structure as the meta-analytic model (Table S1) and that includes the effect sizes' standard errors (SE or sei) as the only moderator (see Equation 21 from the main text). If the slope of this moderator is statistically significant, then we run an additional multilevel meta-regression including the effect sizes' sampling variances (SV or vi) as the only moderator (see Equation 22 from the main text), which allows us to obtain an overall effect (the intercept or B0) after accounting for small-study bias, and this overall is less downwardly  biased when modeling sampling variance than when modeling standard errors (more in the main text; Doucouliagos 2012, 2014).-->

```{r}
# calculating "effective sample size" to account for unbalanced sampling, for SMD and lnRR (esz, hereafter, see Equation 25 from the main manuscript)
# getting the inverse of it

dataset.rr$inv_n_tilda <-  (dataset.rr$C_N + dataset.rr$T_N)/(dataset.rr$C_N*dataset.rr$T_N)
dataset.smd$inv_n_tilda <-  (dataset.smd$C_N + dataset.smd$T_N)/(dataset.smd$C_N*dataset.smd$T_N)


# Application of Equation 27 from the main manuscript
publication.bias.model.RR.srin <- rma.mv(yi, vi,
                                         mods= ~ 1 + sqrt(inv_n_tilda),
                                         random=list(~ 1 | obsID, 
                                                     ~ 1 | paperID),
                                         method="REML",
                                         test = "t", # using t dist rather than z because t is a better, more conservative distribution than z
                                         data=dataset.rr)
#publication.bias.model.RR.srin

publication.bias.model.SMD.srin <- rma.mv(yi, vi,
                                          mods= ~ 1 + sqrt(inv_n_tilda),
                                          random=list(~ 1 | obsID, 
                                                      ~ 1 | paperID),
                                          method="REML",
                                          test = "t", # using t dist rather than z because t is a better, more conservative distribution than z
                                          data=dataset.smd)

#publication.bias.model.SMD.srin

#print(publication.bias.model.1.se,digits=3)

# extracting the mean and 95% confidence intervals
estimates.publication.bias.model.RR.srin <- estimates.CI(publication.bias.model.RR.srin)

estimates.publication.bias.model.SMD.srin <- estimates.CI(publication.bias.model.SMD.srin)

```
OK - once we get rid of an outliner lnRR and SDM match

probably change  below

<!--According to the meta-regression, there is evidence of small-study effects since the slope of the moderator (i.e. SE) is statistically significant (**slope = `r estimates.publication.bias.model.RR.se[2,2]`, 95% CI = [`r estimates.publication.bias.model.1.se[2,3]`,`r estimates.publication.bias.model.RR.se[2,4]`]**; *R<sup>2</sup><sub>marginal</sub>* = `r round(R2(publication.bias.model.RR.se)[1]*100,1)`%; Figure S1), showing that effect sizes with larger SE are larger, and thus, showing evidence that some small effect sizes with large SE are seemingly missing. Since this meta-regression shows some evidence of small-study bias, we  proceed to run the meta-regression suggested in Equation 28 from the main text, so that we can estimate an overall effect size for the meta-analysis after accouting for the existence of small-study bias.-->

**For the time being, I have assumed that the slope for SE is statistically significant, which is not in this dataset, but should be for the final chosen example. That is, even the wording it's based on that, i.e. the results for this specific example do not agree with the description of the results. APOLOGIES if it is confussing**

```{r}
publication.bias.model.RR.se.plot <- predict(publication.bias.model.RR.se)

newdat.RR <- data.frame(sei=dataset.mean.diff$RR.esz.sei,
                        fit=publication.bias.model.RR.se.plot$pred,
                        upper=publication.bias.model.RR.se.plot$ci.ub,
                        lower=publication.bias.model.RR.se.plot$ci.lb,
                        stringsAsFactors=FALSE)

#newdat <- unique(newdat[order(newdat$sei),])
newdat.RR <- unique(newdat.RR[order(newdat.RR$sei),])

xaxis <- dataset.mean.diff$RR.esz.sei
yaxis <- dataset.mean.diff$RR

plot(xaxis,yaxis,
     type="n",
     ylab="",
     xlab="",
     xaxt="n",
     yaxt="n"
     #ylim=c(-1,2),
     #xlim=c(0,1.1)
)


abline(a=0,b=0, lwd=1, lty=1)


axis(1,at=seq(0,1,0.1),
     cex.axis=0.8,tck=-0.02)

axis(2,
     #at=round(seq(-2.5,2.25,0.5),1),
     cex.axis=0.8,las=2,tck=-0.02)

title(xlab = "Effective sample size based standard error (SE)", 
      ylab = "Effect size (lnRR)",
      line = 2.75, cex.lab=1.4)

points(jitter(xaxis,2),yaxis,
       bg=rgb(0,0,0, 0.1),
       col=rgb(0,0,0, 0.2),
       pch=21,
       cex=1)

lines(newdat.RR$sei, newdat.RR$fit, lwd=2.75,col="darkorchid4") 

polygon(c(newdat.RR$sei,rev(newdat.RR$sei)),
        c(newdat.RR$lower,rev(newdat.RR$upper)),
        border=NA,col=rgb(104/255,34/255,139/255, 0.5))
```

**Figure S1.** Effect sizes with larger SE are larger, which provides evidence of small-study effects in this meta-analytic dataset. The solid line represents the model estimate and shading shows the 95% confidence intervals.



##### S4.2.3.2: Time-lag bias test (uni-moderator)

To test for time-lag bias, also called decline effects, we are again going to fit a multilevel meta-regression that has the same random effect structure as the meta-analytic model (Table S1) and that includes the year of publication as the only moderator (see Equation 23 from the main text). The estimated slope for year will tell us whether effect sizes have become smaller over time since the first effect size was published.

```{r}
# Application of Equation 23 from the main manuscript
publication.bias.model.RR.timelag <- rma.mv(yi, vi,
                                            mods= ~ 1 + year,
                                            random=list(~ 1 | obsID, 
                                                        ~ 1 | paperID),
                                            method="REML",
                                            test = "t", # using t dist rather than z because t is a better, more conservative distribution than z
                                            data=dataset.rr)

publication.bias.model.SDM.timelag <- rma.mv(yi, vi,
                                             mods= ~ 1 + year,
                                             random=list(~ 1 | obsID, 
                                                         ~ 1 | paperID),
                                             method="REML",
                                             test = "t", # using t dist rather than z because t is a better, more conservative distribution than z
                                             data=dataset.smd)

#summary(publication.bias.model.1.timelag)

# extracting the mean and 95% confidence intervals
estimates.publication.bias.model.RR.timelag <- estimates.CI(publication.bias.model.RR.timelag)
estimates.publication.bias.model.SMD.timelag <- estimates.CI(publication.bias.model.SDM.timelag)
```

<!--The time-lag bias test indeed shows evidence for a decline effects since the slope of year is negative, meaning that effect sizes have trended towards zero over time (**year slope = `r estimates.publication.bias.model.RR.timelag[2,2]`; 95% CI = [`r estimates.publication.bias.model.RR.timelag[2,3]`,`r estimates.publication.bias.model.RR.timelag[2,4]`]**; *R<sup>2</sup><sub>marginal</sub>* = `r round(R2(publication.bias.model.RR.timelag)[1]*100,1)`%; Figure S2).

```{r}
publication.bias.model.RR.timelag.plot <- predict(publication.bias.model.RR.timelag,newmods=seq(min(dataset.mean.diff$Year),max(dataset.mean.diff$Year),1))

newdat.RR <- data.frame(year=seq(min(dataset.mean.diff$Year),max(dataset.mean.diff$Year),1),
fit=publication.bias.model.RR.timelag.plot$pred,
upper=publication.bias.model.RR.timelag.plot$ci.ub,
lower=publication.bias.model.RR.timelag.plot$ci.lb,
stringsAsFactors=FALSE)


xaxis <- dataset.mean.diff$Year
yaxis <- dataset.mean.diff$RR
cex.study <- 1/dataset.mean.diff$RR.esz.sei

plot(xaxis,yaxis,
type="n",
ylab="",
xlab="",
xaxt="n",
yaxt="n",
ylim=c(-2.25,2.25),
xlim=c(min(xaxis),max(xaxis))
)


abline(a=0,b=0, lwd=1, lty=1)


axis(1, at=seq(min(xaxis),max(xaxis),5),
cex.axis=0.8,tck=-0.02)

axis(2,
#at=round(seq(-2.5,2.25,0.5),1),
cex.axis=0.8,las=2,tck=-0.02)

title(xlab = "Year of publication", 
ylab = "Effect size (lnRR)",
line = 2.75, cex.lab=1.4)

points(jitter(xaxis,2),yaxis,
bg=rgb(0,0,0, 0.1),
col=rgb(0,0,0, 0.2),
pch=21,
cex=cex.study)

lines(newdat.RR$year, newdat.RR$fit, lwd=2.75,col="darkorchid4") 

polygon(c(newdat.RR$year,rev(newdat.RR$year)),
c(newdat.RR$lower,rev(newdat.RR$upper)),
border=NA,col=rgb(104/255,34/255,139/255, 0.5))
```

**Figure S2.** The overall published effect size has decreased over time since first described. The solid line represents the model estimate, shading shows the 95% confidence intervals and individual effect sizes are scaled by their precision (1/SE).-->

<br/><br/>


##### S4.2.3.3:  All-in publication bias test (mulit-moderators)

Try not to repeat what you do not need to 

<!--When heterogeneity exists, it is best to combine Equations 27 and 23 with other moderators since those additional moderators will generally explain or be predicted to explain some of the heterogeneity among effect sizes. Therefore, in this case we will run a multilevel meta-regression including the effective sample size based standard errors, the year of publication and the following moderators originally included in XXX et al. (XXXX): 'season', 'group composition' and 'type of interactions'.-->

```{r}
# preparing the moderators that need to be included in a meta-regression that also contains a  moderator with the standard errors of the effect sizes and the year of publication

# Application of Equation 29 from the main manuscript
publication.bias.model.RR.timelag.srin <- rma.mv(yi, vi,
                                                 mods= ~ 1 + sqrt(inv_n_tilda) + 
                                                   scale(year) + 
                                                   Experimental.or.Observational.,
                                                 random=list(
                                                   ~1 |Disturbance_category,
                                                   ~ 1 | obsID, 
                                                   ~ 1 | paperID),
                                                 method="REML",
                                                 test = "t", # using t dist rather than z because t is a better, more conservative distribution than z
                                                 data=dataset.rr,
                                                 control=list(optimizer="optim", optmethod="Nelder-Mead"))

###
publication.bias.model.RR.timelag.all.eN <- rma.mv(yi, vi,
                                                   mods= ~ 1 + inv_n_tilda + 
                                                     scale(year) + 
                                                     Experimental.or.Observational.,
                                                   random=list(
                                                     ~1 |Disturbance_category,
                                                     ~ 1 | obsID, 
                                                     ~ 1 | paperID),
                                                   method="REML",
                                                   test = "t", # using t dist rather than z because t is a better, more conservative distribution than z
                                                   data=dataset.rr,
                                                   control=list(optimizer="optim", optmethod="Nelder-Mead"))

publication.bias.model.RR.timelag.all.eN1 <- rma.mv(yi, vi,
                                                    mods= ~ 1 + inv_n_tilda + 
                                                      scale(year),
                                                    random=list(
                                                      ~1 |Disturbance_category,
                                                      ~ 1 | obsID, 
                                                      ~ 1 | paperID),
                                                    method="REML",
                                                    test = "t", # using t dist rather than z because t is a better, more conservative distribution than z
                                                    data=dataset.rr,
                                                    control=list(optimizer="optim", optmethod="Nelder-Mead"))
#####

# Application of Equation 29 from the main manuscript
publication.bias.model.SMD.timelag.al.srin <- rma.mv(yi, vi,
                                                     mods= ~ 1 + sqrt(inv_n_tilda) + 
                                                       scale(year) + 
                                                       Experimental.or.Observational.,
                                                     random=list(
                                                       ~1 |Disturbance_category,
                                                       ~ 1 | obsID, 
                                                       ~ 1 | paperID),
                                                     method="REML",
                                                     test = "t", # using t dist rather than z because t is a better, more conservative distribution than z
                                                     data=dataset.smd,
                                                     control=list(optimizer="optim", optmethod="Nelder-Mead"))

publication.bias.model.SMD.timelag.al.eN <- rma.mv(yi, vi,
                                                   mods= ~ 1 + inv_n_tilda + 
                                                     scale(year) + 
                                                     Experimental.or.Observational.,
                                                   random=list(
                                                     ~1 |Disturbance_category,
                                                     ~ 1 | obsID, 
                                                     ~ 1 | paperID),
                                                   method="REML",
                                                   test = "t", # using t dist rather than z because t is a better, more conservative distribution than z
                                                   data=dataset.smd,
                                                   control=list(optimizer="optim", optmethod="Nelder-Mead"))

publication.bias.model.SMD.timelag.al.eN1 <- rma.mv(yi, vi,
                                                    mods= ~ 1 + inv_n_tilda + 
                                                      scale(year),
                                                    random=list(
                                                      ~1 |Disturbance_category,
                                                      ~ 1 | obsID, 
                                                      ~ 1 | paperID),
                                                    method="REML",
                                                    test = "t", # using t dist rather than z because t is a better, more conservative distribution than z
                                                    data=dataset.smd)


```

**Need to add a little text and probably also a table showing the results of this meta-regresssion, which I did not do because it would be very specific for the dataset we will finally used. Nonetheless, find below the *R<sup>2</sup><sub>marginal</sub>* estimate, which for this sparrow example is pretty high and much coming from the publication bias tests**

*R<sup>2</sup><sub>marginal</sub>* = `r round(R2(publication.bias.model.1.timelag.all.in)[1]*100,1)`%


## Appendix S5: Avearging and sampling approches for publication bias

Some intro - an overview 

1. averaging vs sampling?
2. what is trim and fill methods?
3. what is 3 PSM?

### S5.1: Correlation effect sizes (**Zr**)

#### S5.1.1: Averaging 

use aggregate

```{r}
# rho = 0.5 is as in Noble et al (Mol Ecol)
dataset.r <- escalc(yi=yi, vi=vi, data=dataset.r)
dataset.r.agg <- aggregate(dataset.r, cluster=study, struct="CS", rho = 0.5)
dataset.r.agg

```


##### S5.1.1.2: Trim and fill test 

This may not work but do this anyways

```{r}

meta.analysis.model.r.agg <- rma(yi, vi, test="knha", # using t dist rather than z
                                 data=dataset.r)
trimfill.r <- trimfill(meta.analysis.model.r.agg)
funnel(trimfill.r, yaxis="seinv")

```

##### S5.1.1.3: Selection model (3 PSM)

```{r}
# Note we need to set  method="ML",
# withou modrators

meta.regression.r.agg0 <- rma(yi, vi, 
                              method="ML",
                              test="knha", 
                              data=dataset.r)


three.PSM.r0 <- selmodel(meta.regression.r.agg0, type="stepfun", steps=c(0.05, 1))

# look at confidence intervals - not just point estimate
summary(three.PSM.r0)


# with moderators 
meta.regression.r.agg <- rma(yi, vi, 
                             mod = ~ scale(year) 
                             + CaptivityC - 1, 
                             #+ Class-1,
                             method="ML",
                             test="knha", 
                             data=dataset.r)


three.PSM.r <- selmodel(meta.regression.r.agg, type="stepfun", steps=c(0.05, 1))

summary(three.PSM.r)
```


##### S5.1.1.4: Cumlative meta-analysis 

I guess we do not expect a decline effect - but we do this anyways

Easy.....
```{r}
cum.meta.analysis.model.r.agg <- cumul(meta.analysis.model.r.agg)
forest(cum.meta.analysis.model.r.agg, xlab = "Overall estimate (Zr)")
```

#### S5.1.2: Sampling 

##### S5.1.2.2: Trim and fill test 


I will leave this to Alfredo and Yefeng to solve (but discuss it with them to give some directions)

##### S5.1.2.3: Selection model (3 PSM)

##### S5.1.2.4: Cumlative meta-analysis 

### S5.2: Mean differences between two groups (lnRR and SMD)

#### S5.2.1: Averaging 
```{r}
# rho = 0.5 is as in Noble et al (Mol Ecol)
dataset.rr <- escalc(yi=yi, vi=vi, data=dataset.rr)
dataset.rr.agg <- aggregate(dataset.rr, cluster=paperID, struct="CS", rho = 0.5)
dataset.rr.agg

dataset.smd <- escalc(yi=yi, vi=vi, data=dataset.smd)
dataset.smd.agg <- aggregate(dataset.smd, cluster=paperID, struct="CS", rho = 0.5)
dataset.smd.agg

```


##### S5.2.1.2: Trim and fill test 

```{r}

# lnRR
meta.analysis.model.rr.agg <- rma(yi, vi, test="knha", # using t dist rather than z
                                  data=dataset.rr.agg)
trimfill.rr <- trimfill(meta.analysis.model.rr.agg)

#SMD
meta.analysis.model.smd.agg <- rma(yi, vi, test="knha", # using t dist rather than z
                                   data=dataset.smd.agg)
trimfill.smd <- trimfill(meta.analysis.model.smd.agg)

par(mfrow = c(1,2))
funnel(trimfill.rr, yaxis="seinv",
       ylab = "Precision (1/SE)",
       xlab = "Effect size (lnRR)") 
funnel(trimfill.smd, yaxis="seinv",
       ylab = "Precision (1/SE)",
       xlab = "Effect size (SDM)") 

```

##### S5.2.1.3: Selection model (3 PSM)

```{r}

# lnRR

# moderator
meta.regression.rr.agg0 <- rma(yi, vi, 
                               #mod = ~ scale(year),
                               method="ML",
                               test="knha", 
                               data=dataset.rr)


three.PSM.rr0 <- selmodel(meta.regression.rr.agg0, type="stepfun", steps=c(0.05, 1))

summary(three.PSM.rr0)

# taking out Exp vs Obs
meta.regression.rr.agg0 <- rma(yi, vi, 
                               #mod = ~ scale(year),
                               method="ML",
                               test="knha", 
                               data=dataset.rr)


three.PSM.rr0 <- selmodel(meta.regression.rr.agg0, type="stepfun", steps=c(0.05, 1))

summary(three.PSM.rr0)

# with moderator
meta.regression.rr.agg <- rma(yi, vi, 
                              mod = ~ scale(year) 
                              + Experimental.or.Observational. - 1, 
                              #+ Class-1,
                              method="ML",
                              test="knha", 
                              data=dataset.rr)


three.PSM.rr <- selmodel(meta.regression.rr.agg, type="stepfun", steps=c(0.05, 1))

summary(three.PSM.rr)


## SMD

# without moderators
meta.regression.smd.agg0 <- rma(yi, vi, 
                                method="ML",
                                test="knha", 
                                data=dataset.smd)


three.PSM.smd0 <- selmodel(meta.regression.smd.agg0, type="stepfun", steps=c(0.05, 1))

summary(three.PSM.smd0)


# with moderators
meta.regression.smd.agg <- rma(yi, vi, 
                               mod = ~ scale(year) 
                               + Experimental.or.Observational. - 1, 
                               #+ Class-1,
                               method="ML",
                               test="knha", 
                               data=dataset.smd)


three.PSM.smd <- selmodel(meta.regression.smd.agg, type="stepfun", steps=c(0.05, 1))

summary(three.PSM.smd)

```


##### S5.2.1.4: Cumlative meta-analysis 

```{r, fig.height= 8, fig.width= 7}
# lnRR
cum.meta.analysis.model.rr.agg <- cumul(meta.analysis.model.rr.agg)

# SMD
cum.meta.analysis.model.smd.agg <- cumul(meta.analysis.model.smd.agg)

par(mfrow = c(1, 2))

forest(cum.meta.analysis.model.rr.agg, xlab = "Overall estimate (lnRR)")
forest(cum.meta.analysis.model.smd.agg, xlab = "Overall estimate (SMD)")
```

#### S5.2.2: Sampling 

##### S5.2.2.2: Trim and fill test 

##### S5.2.2.3: Selection model (3 PSM)

##### S5.2.2.4: Cumlative meta-analysis 

## Appendix extra: Figure's R code

##### Code for Fig. 1

There is no code for Fig. 1 (created in PowerPoint)

##### Code for Fig. 2

Rose??

##### Code for Fig. 3

```{r, fig.height= 8, fig.width= 7}

# creating data

set.seed(77777)

# setting parameters
n.effect <- 100
sigma2.s <- 0.05
beta1 <- 0.2
# using negative binomial we get good spread of sample size
n.sample <- rnbinom(n.effect, mu = 30, size = 0.7) + 4
# variance for Zr
vi <- 1/(n.sample - 3)
# moderator x 1
xi <- rnorm(n.effect)

# there is underling overall effect to r = 0.2 or Zr = 0.203
Zr <- atanh(0.2) + beta1*xi + rnorm(n.effect, 0, sqrt(sigma2.s)) + rnorm(n.effect, 0, sqrt(vi))
#qplot(Zr, 1/sqrt(vi))

dat <- data.frame(yi = Zr, vi = vi, sei = sqrt(vi), xi = xi, ni = n.sample, prec = 1 / sqrt(vi), wi = 1/vi, zi = Zr/sqrt(vi))

rows <- 1:nrow(dat)
expected <- which(1/dat$sei < 5 & dat$yi < 0.25)
unexpected <- which(1/dat$sei > 4.7 & dat$yi > 0.25)

col_point <- ifelse(rows %in% expected, "red", ifelse(rows %in% unexpected, "blue", "black"))

dat$col_point <- col_point

#https://stackoverflow.com/questions/50012553/combining-plots-created-by-r-base-lattice-and-ggplot2
# saving base plot as an object!!! https://www.andrewheiss.com/blog/2016/12/08/save-base-graphics-as-pseudo-objects-in-r/

mod_ra <- rma(yi, vi, data = dat)
mod_fe <- rma(yi, vi, data = dat, method = "FE")

# margin controlling
#par(mar=c(5,4,0,0) + 0.1)

# - done TODO - DL for sunset plot and enhanced counter - making it - 0.01 < p < 0.05 

# a
pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
funnel(dat$yi, dat$vi, ni = dat$ni, yaxis="ni",
       xlim = c(-3, 3),
       refline=mod_ra$beta, xlab = "Effect size (Zr)") 
a <- recordPlot()
invisible(dev.off())


# b
pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
funnel(mod_ra, yaxis="seinv", col = col_point,  
       ylim = c(1, 12), xlim = c(-3, 3),
       ylab = "Precison (1/SE)", xlab = "Effect size (Zr)")
legend(x = 1.4, y = 12, legend = c("expected", "","unexpected"), pch = c(19, 0, 19), col = c("red", "grey83", "blue"), bty = "n")
b <- recordPlot()
invisible(dev.off())


# c
pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
funnel(mod_ra, 
       yaxis = "sei", 
       xlim = c(-3, 3),
       level = c(95, 99), 
       shade = c("white", "gray55"), 
       refline = 0, legend = F, 
       ylab = "Standard error (SE)", xlab = "Effect size (Zr)")
legend(x = 1, y = 0, legend = c("0.01 < p < 0.05"), pch = c(15), col = c("gray55"), bty = "n")
c <- recordPlot()
invisible(dev.off())

# d
d <- viz_sunset(dat[,c("yi", "sei")], 
                power_contours = "continuous", 
                contours = T,
                power_stats = F,
                xlab = "Effect size (Zr)",
                ylab = "Standard error (SE)")


# e
# meta-regression (random-effects model)
mod_re1 <- rma(yi, vi, mod = ~ xi, data = dat)


pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
funnel(mod_re1, 
       yaxis = "sei", 
       xlim = c(-3, 3), ylim = c(0, 1),
       #level = c(95, 99), 
       #shade = c("white", "gray75"), 
       refline = 0, legend = F,  ylab = "Standard error (SE)", xlab = "Standardized residuls (Zr)")
#legend(x = 1, y = 0, legend = c("0.01 < p < 0.05", ""), pch = c(15, 0), col = c("gray75", "white"), bg = "white")
e <- recordPlot()
invisible(dev.off())

# f

# 
mod <- metagen(TE = yi, seTE = sqrt(vi) , data = dat)

pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
#radial.rma(mod_fe, xlim = c(0, 12.5), zlab = "Z score", xlab = "Precision (1/SE)") 
#legend(1,1, "test", bty = "n")
radial(mod, level = 0.95, pch = 19,  xlab = "Precision (1/SE)", ylab = "Z score")
#abline(a = 0, b = 0.1908, lwd = 1.5)
f <- recordPlot()
invisible(dev.off())

# fig 3
(ggdraw(a) + ggdraw(b)) /(ggdraw(c) + ggdraw(d))/ (plot_grid(e) + ggdraw(f)) + plot_annotation(tag_levels = 'a')

```

Fig 3 - see the text

##### Code for Fig. 4

```{r, fig.height= 8, fig.width= 8}
# creating data with publication bias

dat2 <- dat[dat$col_point != "red", ]

# meta-analysis
mod_ra2 <- rma(yi, vi, data = dat2)

# egger regression (the same as in S2)
egger1 <- lm(zi ~ prec, data = dat2)
egger2 <- lm(yi ~ sei, weight = wi, data = dat2)

# data for time-lag bias
set.seed(123)
position <- sample(1:nrow(dat2), size = 15)

dat3 <- dat2[sort(position), ]

sorting <- c(12, 13, 14, 3, 4, 9, 8, 5, 10, 11,  6, 7, 2, 15, 1)

dat3 <- dat3[sorting, ]        
dat3$year <- c(2000, 2001, 2003, 2005, 2006, 2007, 2008, 2009, 2010, 2012, 2013, 2014, 2016, 2018, 2019)

# cumlatie meta-analysis
cum <- rma(yi, vi, data=dat3)
cum_ma <- cumul(cum)
#forest(cum_ma, xlab = "Overall estimate (Zr)")


# meta-regression

mod_re2 <- rma(yi, vi, mod = ~ year, data = dat3)

# funnel
taf <- trimfill(mod_ra2)

# Plotting from here

p1 <-  ggplot(dat2, aes(prec, zi)) + geom_point() +
  geom_hline(yintercept = 0, linetype = "dashed") + 
  geom_hline(yintercept = egger1$coefficients[[1]]) +
  #geom_smooth(method = "lm") + 
  labs(x = "Precision (1/SE)", 
       y = "Z score")

p2 <- ggplot(dat2, aes(sei, yi)) + geom_point() +
  # geom_smooth(method = "lm") + 
  geom_hline(yintercept = 0, linetype = "dashed") + 
  geom_abline(intercept = egger2$coefficients[[1]], slope = egger2$coefficients[[2]]) +
  labs(x = "Standard error (SE)", 
       y = "Effect size (Zr)")

# c and d
# probably 15 data points for cumulative meta-analyses

cum <- rma(yi, vi, data=dat3)
cum_ma <- cumul(cum)

# p3
pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
forest(cum_ma, xlab = "Overall estimate (Zr)")
p3 <- recordPlot()
invisible(dev.off())

# p4
p4 <- ggplot(dat3, aes(x = year, y = yi, size = prec)) +
  geom_hline(yintercept = 0, linetype = "dashed") + 
  geom_abline(intercept = mod_re2$beta[[1]], slope = mod_re2$beta[[2]]) +
  geom_point(shape = 21, fill = "grey90") + 
  labs(x = "Publication year", y = "Effect size (Zr)", size ="Precision (1/SE)") + 
  guides(fill = "none", colour = "none") +
  theme(legend.position = c(0, 0.15), legend.justification = c(0, 1)) +
  theme(legend.direction = "horizontal") +
  # theme(legend.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.background = element_blank()) +
  theme(axis.text.y = element_text(size = 10, colour = "black", hjust = 0.5, angle = 90))

# p5
pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
funnel(taf, yaxis="seinv", col = col_point,  estimator="R0",
       ylim = c(1, 12), 
       #xlim = c(-3, 3),
       ylab = "Precison (1/SE)", xlab = "Effect size (Zr)")
p5 <- recordPlot()
invisible(dev.off())

# p6
pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
funnel(taf, yaxis="sei", col = col_point,  estimator="R0",
       #xlim = c(-3, 3),
       ylab = "Standard error (SE)", xlab = "Effect size (Zr)")
p6 <- recordPlot()
invisible(dev.off())


# e and f
# maybe use the same datasets as Fig 3

# fig 4
(plot_grid(p1) + plot_grid(p2)) /(ggdraw(p3) + plot_grid(p4))/(plot_grid(p5) + plot_grid(p6)) + plot_annotation(tag_levels = 'a')


```


##### Code for Fig. 5

```{r, fig.height= 8, fig.width= 7, message = FALSE}
# dmetr

#devtools::install_github("MathiasHarrer/dmetar")
#library(dmetar)

p_dat <- data.frame("studlab" = as.factor(1:nrow(dat2)), "TE" = dat2$yi, "seTE" = dat2$sei)
#pcurve(p_dat)

pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
invisible(pcurve(p_dat))
plot<- recordPlot()
invisible(dev.off())

# turuning the base plot into ggplot
plot <- ggdraw(plot)

#dat2

model <- rma(yi, vi = vi, data = dat2, method = "ML", mod = ~ xi)

# only sel1 and sel3 are used
#sel0 <- selmodel(model, type = "beta")
sel1 <- selmodel(model, type="halfnorm", prec="sei", scaleprec=FALSE)
#sel2 <- selmodel(model, type="negexp",   prec="sei", scaleprec=FALSE)
sel3 <- selmodel(model, type="logistic", prec="sei", scaleprec=FALSE)
#sel4 <- selmodel(model, type="power",    prec="sei",  scaleprec=FALSE)
sel1b <- selmodel(model, type="halfnorm")
#sel2b <- selmodel(model, type="negexp")
sel3b <- selmodel(model, type="logistic")
#sel4b <- selmodel(model, type="power")

# step function 
sel5 <- selmodel(model, type="stepfun", steps=c(0.05, 0.10, 0.50, 1.00))
#sel6 <- selmodel(model, type="stepfun", steps=c(0.001, 0.01, 0.05, 0.10, 0.25, 0.50, 0.75, 1))
sel7 <- selmodel(model, type="stepfun", steps=c(0.05, 1))

#plot(sel5, ylim = c(0,1))
#plot(sel6, add=TRUE, col="red", scale = T)
#plot(sel7, add=TRUE, col="red")
#legend(x = 0.5, y = 1, legend = c("3 cutpoints", "","1 cutpoint (3 PMS)"), lty =c(1, 0, 1), col = c("black", "", "red"), bty = "n")

# plotting from here
pA <- plot +  theme(plot.margin=unit(c(-1,0,-1,0),"cm"))

# B
pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
plot(sel1)
plot(sel3, add=TRUE, col="red")
plot(sel1b, add=TRUE, lty = "dotted")
plot(sel3b, add=TRUE, col="red", lty = "dotted")
legend(x = 0.6, y = 1, legend = c("half-normal", "","logistic"), lty =c(1, 0, 1), col = c("black", "", "red"), bty = "n")
pB <- recordPlot()
invisible(dev.off())

# C
pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
plot(sel5, ylim = c(0,1))
plot(sel7, add=TRUE, col="red", lty = 3)
legend(x = 0.5, y = 1, legend = c("3 cutpoints", "","1 cutpoint (3 PSM)"), lty =c(1, 0, 1), col = c("black", "", "red"), bty = "n")
pC <- recordPlot()
invisible(dev.off())

# Fig 5
pA /(ggdraw(pB) + ggdraw(pC)) + plot_annotation(tag_levels = 'a')

```

##### Code for Fig. 6

```{r, fig.height= 8, fig.width= 7, message = FALSE}

# Zr data creation
set.seed(777)
# setting parameters
n.study  <- 70
sigma2.s <- 0.05
sigma2.u <- 0.05
beta1 <- 0.1
beta2 <- - 0.05
# using negative binomial we get good spread of sample size
n.sample <- rnbinom(n.study, mu = 30, size = 0.2) + 4

# the number of effect size per study
set.seed(777777)
n.es.per.s <- rpois(n.study,3) + 1
#sum(n.es.per.s)

n.effect <- sum(n.es.per.s)
# variance for Zr
vi <- 1/(rep(n.sample, n.es.per.s) - 3)

# In the text, we have two moderators
# moderator x 1 - study level
x1j <- rnorm(n.study)
# moderator x 2 - effect isze level
x2i <- rnorm(n.effect)

# study position
place <- rep(1:n.study, n.es.per.s)

# there is underling overall effect to r = 0.3 or Zr = 0.3095196
Zr <- 0 + rnorm(n.study, 0, sqrt(sigma2.s))[place] + 
  rnorm(n.effect, 0, sqrt(sigma2.u)) + rnorm(n.effect, 0, sqrt(vi))

#qplot(Zr, 1/sqrt(vi))

datA <- data.frame(yi = Zr, vi = vi, sei = sqrt(vi), x1 = x1j[place], x2 = x2i, ni = n.sample[place], prec = 1 / sqrt(vi), wi = 1/vi, zi = Zr/sqrt(vi), studyID = place, effectID = 1:n.effect)


# cutting publication mbias like stuff
expected <- c(which(1/datA$sei < 5.5 & datA$yi < 0.2), which(1/datA$sei < 3 & datA$yi < 0.7))

datB <- datA[-expected, ]

#qplot(x = yi, y = 1/sqrt(vi), data = datB)

mod_ma <- rma.mv(yi, vi, mod = ~ sqrt(vi), random = list(~ 1 | studyID, ~1 | effectID), data = datB)

mod_mb <- rma.mv(yi, vi, mod = ~ vi, random = list(~ 1 | studyID, ~1 | effectID), data = datB)

#summary(mod_ma)
#summary(mod_mb)


# creating curve lines
x = seq(0, 1, by = 0.001)
#x <- x[-1000]
y =  mod_mb$beta[[1]] + sqrt(x)* mod_mb$beta[[2]]

vi_curve <- tibble(x = x, y = y)

# lnRR

set.seed(777)
# setting parameters
n.study  <- 30
sigma2.s <- 0.01
sigma2.u <- 0.005
beta1 <- 0.1
beta2 <- -0.01
# using negative binomial we get good spread of sample size
n.sample <- rnbinom(n.study, mu = 40, size = 0.5) + 4

# the number of effect size per study
set.seed(7777)
n.es.per.s <- rpois(n.study,3) + 1
#sum(n.es.per.s)

# sample size
n.effect <- sum(n.es.per.s)

# cv for each study

cv <- sample(seq(0.3, 0.35,by = 0.001), n.effect, replace = T)

rep_sample <- rep(n.sample, times = n.es.per.s)

# variance for lnRR
vi <- (2*cv^2)/rep_sample

# In the text, we have two moderators
# moderator x 1 - study level
x1j <- rnorm(n.study)
# moderator x 2 - effect isze level
x2i <- rnorm(n.effect)

# study position

pos <- rep(1:n.study, n.es.per.s)

# there is underling overall effect to lnRR = 0.2

dat3 <- data.frame(vi = vi, sei = sqrt(vi), x1 = x1j[pos], x2 = x2i, ni = n.sample[pos], prec = 1 / sqrt(vi), wi = 1/vi, studyID = pos, effectID = 1:n.effect)

# M matrix (V = VCV)
sigma <- make_VCV_matrix(dat3, "vi", "studyID", "effectID", rho = 0)

lnRR<- 0.3 + beta1*x1j[pos] + beta2*x2i + rnorm(n.study, 0, sqrt(sigma2.s))[pos] + 
  rnorm(n.effect, 0, sqrt(sigma2.u)) +  MASS::mvrnorm(1, rep(0, n.effect), sigma)

dat3$yi <- lnRR

mod_ml <- rma.mv(yi, vi, mod = ~ x1 + x2, random = list(~ 1 | studyID, ~1 | effectID), data = dat3)

# residuals - stadnarised
#summary(mod_ml)
#funnel(mod_ml, yaxis="seinv", col = inferno(25)[pos])

# residual marginal
residm <- dat3$yi - predict(mod_ml)[[1]]
vi_residm <- vi + predict(mod_ml)[[2]]^2 # adding prediction errors 


# residual conditional 1 & 2 + its error
blups <- ranef(mod_ml)

study_effect <- blups$studyID[[1]][pos]
vi_study <- (blups$studyID[[2]][pos])^2

residc1 <- dat3$yi - (predict(mod_ml)[[1]] + study_effect)
vi_residc1 <- vi_residm + vi_study

effect_effect <- blups$effectID[[1]]
vi_effect <- blups$effectID[[1]]^2

residc2 <- dat3$yi - (predict(mod_ml)[[1]] + study_effect + effect_effect)
vi_residc2 <- vi_residm + vi_study + vi_effect


# plotting
# Panel A
mod1 <- rma(yi, vi, data = dat3)
pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
funnel(mod1, col =inferno(25)[pos],
       ylim = c(0, 0.26), xlim = c(-0.6, 0.8),
       ylab = "Standard error (SE)", xlab = "Effect size (lnRR)")
pa <- recordPlot()
invisible(dev.off())

# Panel B
pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
funnel (residm, vi = vi_residm, yaxis = "sei",  col = inferno(25)[pos],  
        ylim = c(0, 0.26), xlim = c(-0.6, 0.8),
        ylab = "Standard error (SE)", xlab = "Residuals marginal (lnRR)")
pb <- recordPlot()
invisible(dev.off())

# Panel C
pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
funnel(residc1, vi = vi_residc1, yaxis = "sei",  col = inferno(25)[pos],
       ylim = c(0, 0.26), xlim = c(-0.6, 0.8),
       ylab = "Standard error (SE)", xlab = "Residuals conditional 1 (lnRR)")
pc <- recordPlot()
invisible(dev.off())

# Panel D
pdf(NULL)
dev.control(displaylist="enable")
par(mar=c(4,4,0.1,0) + 0.1)
funnel(residc2, vi = vi_residc2, yaxis = "sei",  col = inferno(25)[pos],
       ylim = c(0, 0.26), xlim = c(-0.6, 0.8),
       ylab = "Standard error (SE)", xlab = "Residuals conditional 2 (lnRR)")
pd <- recordPlot()
invisible(dev.off())

# Panel E
pe <- ggplot(datB, aes(sei, yi)) + geom_point() + 
  geom_line(data = vi_curve, mapping = aes(x = x, y = y, col = "red")) +
  #geom_point(aes(col = studyID)) +
  # geom_smooth(method = "lm") + 
  geom_hline(yintercept = 0, linetype = "dashed") + 
  geom_abline(intercept = mod_ma$beta[[1]], slope = mod_ma$beta[[2]]) +
  xlim(0, 1) + ylim(-1.5, 2.5) + 
  labs(x = "Standard error (SE)", y = "Effect size (Zr)")+ 
  theme(legend.position = "none")
# F

# Variance
pf <- ggplot(datB, aes(vi, yi)) + geom_point() + 
  #geom_point(aes(col = studyID)) +
  # geom_smooth(method = "lm") + 
  geom_hline(yintercept = 0, linetype = "dashed") + 
  geom_abline(intercept = mod_mb$beta[[1]], slope = mod_mb$beta[[2]], col = "red") +
  xlim(0, 1) + ylim(-1.5, 2.5) + 
  labs(x = "Sampling variance", y = "Effect size (Zr)")

# fig6
(ggdraw(pa) + ggdraw(pb)) /(ggdraw(pc) + ggdraw(pd))/(plot_grid(pe) + plot_grid(pf)) + plot_annotation(tag_levels = 'a')

```
##### Code for Fig. 7

There is no code for Fig. 7 (created in PowerPoint)


## R Session Information

```{r}
sessionInfo() %>% pander()
```

## References

To be added.